### PR TITLE
Environment detection

### PR DESCRIPTION
**Pull request by gbirke**
https://phabricator.wikimedia.org/T351627

### CODE REVIEW

**Comment by gbirke** [comment]
The new dev-mode-aware getImpressions is currently only used in the Desktop. I'd like to wait for approval/change requests before I change all the other banners.

**Comment by Abban** [comment]
LGTM

File: src/utils/RuntimeEnvironment.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+}
+
+export interface RuntimeEnvironment {
+	isInDevMode: boolean;
+	runsInDevEnvironment: boolean;
+}
+
+export class UrlRuntimeEnvironment implements RuntimeEnvironment {
+	private _isInDevMode: boolean;
+	private _runsInDevEnvironment: boolean;
```
Comment (moiikana) [inline src/utils/RuntimeEnvironment.ts:16]: what are the different purposes of these two fields?

Comment (Abban) [reply src/utils/RuntimeEnvironment.ts:16]: One marks the `?devMode` query for previewing the banners on WPORG, the other is the `?devBanner` query item from the Fun Forge

**Review by Abban — APPROVED** [review]

**Pull request merged**